parser.go

```diff
@@ -0,0 +1,345 @@
+package proteus
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/simplesurance/proteus/internal/consts"
+	"github.com/simplesurance/proteus/specialflags"
+	"github.com/simplesurance/proteus/types"
+)
+
+// MustParse reads the parameters into the provided structure reference. The
+// provided parameters struct can be annotated with some parameter tags to
+// configure how the configuration is read.
+//
+// There are support for sub-parameters and for getting updates about
+// changes in value without the need to restart the application. See godoc
+// examples for usage.
+//
+// A Parsed object is guaranteed to be always returned, even in case of error,
+// allowing to get usage information.
+func MustParse(parameters any, options ...Option) (*Parsed, error) {
+	opts := settings{
+		loggerFn: func(msg string, depth int) {}, // nop logger
+	}
+	opts.apply(options...)
+
+	appConfig, err := mustInferConfigFromValue(parameters, opts)
+	if err != nil {
+		panic(fmt.Errorf("INVALID CONFIGURATION STRUCT: %v", err))
+	}
+
+	if len(opts.sources) == 0 {
+		panic("NO CONFIGURATION SOURCE WAS PROVIDED")
+	}
```

Comment:
Those two panic conditions would only happen in case of coding error, where the caller is not following the contract the method requires. Returning an error would imply in returning an invalid Parsed object, since the error happened while creating it. In my opinion this is the same as passing a nil pointer to a function that expects a non-nil value.